substrate-node/pallets/pallet-smart-contract/src/lib.rs

```diff
@@ -1297,20 +1301,11 @@ impl<T: Config> Pallet<T> {
 
     fn handle_lock(
         contract: &mut types::Contract<T>,
+        mut contract_lock: types::ContractLock<BalanceOf<T>>,
```

Comment:
Taking the lock as an argument is fine to avoid looking it up twice, but this should be changed since it is also inserted twice. Furthermore, changes to the lock here will be ignored, since they are saved, but the original object is not returned, and the saved value is later overwritten in the caller